### ISSUE TITLE
Game crashing while stopping download

### ISSUE THREAD

**Issue by IsNevis** [Pokechu22/WorldDownloader#239]
Log: [latest.log](https://github.com/Pokechu22/WorldDownloader/files/7831225/latest.log)
Debug(? idk this was in the logs folder too): [debug.log](https://github.com/Pokechu22/WorldDownloader/files/7831226/debug.log)
Crash: [crash-2022-01-08_02.03.49-client.txt](https://github.com/Pokechu22/WorldDownloader/files/7831228/crash-2022-01-08_02.03.49-client.txt)

**Comment by Pokechu22** [comment]
Looks like it crashed trying to save the biomes for the world.  Biomes are fairly complicated in 1.16+, as they are data-driven and can be defined by the server.  In this case, it failed because there was no nether biome, and thus the normal world generation settings (which expect there to be a nether) broke.  Is this is a skyblock server with no nether dimension at all?

**Comment by IsNevis** [comment]
Yeh it's a Skyblock server. And yeh pretty sure there is no nether

**Comment by Pokechu22** [comment]
If you have the ability to connect to the server with older Minecraft versions, try using 1.16.1 or older (for 1.16.1 you'll have to install it as a base edit using the "add to minecraft.jar" functionality in MultiMC).  That version predates the dynamic biome system (which was added in 1.16.2 if I recall correctly) so the crash shouldn't happen.

**Comment by IsNevis** [comment]
Well gonna try it. If it works it will be perfect as the server support starts from 1.16.1

**Comment by IsNevis** [comment]
ok it works. Thanks for the help. 

I think you should fix this issue next time you update : )

**Comment by Pokechu22** [comment]
Yeah, I'll look into it - biomes in general are complicated and something I haven't had time to investigate, but crashing is definitely something that shouldn't happen.

**Comment by IsNevis** [comment]
you could maybe add an option to ignore biomes so it doesn't crashes. well thats a simple solution

**Comment by FSWei** [comment]
thank you~
